mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAO.java

```diff
@@ -334,7 +335,11 @@ private PreparedStatement prepareSelectUidRangeLimited(Session session) {
     private Optional<ComposedMessageIdWithMetaData> fromRowToComposedMessageIdWithFlags(Row row) {
         if (row.getUUID(MESSAGE_ID) == null) {
             // Out of order updates with concurrent deletes can result in the row being partially deleted
-            // We filter out such records
+            // We filter out such records, and cleanup them.
+            delete(CassandraId.of(row.getUUID(MAILBOX_ID)),
+                MessageUid.of(row.getLong(IMAP_UID)))
+                .subscribeOn(Schedulers.elastic())
+                .subscribe();
```

Comment:
thanks! but is it frequent to subscribe to a scheduler inside of a Flux?

Reply:
Yes, we do it eg on the MailQueue to perform some async operations. (We want to trigger it, we do not care about its completion.)